src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/Value.ReadTests.cs

```diff
@@ -442,5 +448,66 @@ private static void DeserializeLongJsonString(int stringLength)
             string str = JsonSerializer.Deserialize<string>(json);
             Assert.True(json.AsSpan(1, json.Length - 2).SequenceEqual(str.AsSpan()));
         }
+
+        [Theory]
+        [InlineData("23:59:59")]
+        [InlineData("\\u0032\\u0033\\u003A\\u0035\\u0039\\u003A\\u0035\\u0039", "23:59:59")]
+        [InlineData("23:59:59.9", "23:59:59.9000000")]
+        [InlineData("23:59:59.9999999")]
+        [InlineData("9999999.23:59:59.9999999")]
+        [InlineData("-9999999.23:59:59.9999999")]
+        [InlineData("10675199.02:48:05.4775807")] // TimeSpan.MaxValue
+        [InlineData("-10675199.02:48:05.4775808")] // TimeSpan.MinValue
+        public static void TimeSpan_Read_Success(string json, string? actual = null)
```

Comment:
If you're relying on `Utf8Parser.TryParse(..., out TimeSpan, ...)`, please also add these unit tests:

```cs
[InlineData("\\t23:59:59")] // should succeed since Utf8Parser.TryParse drops leading horizontal tab chars
[InlineData("\t23:59:59")] // should fail since illegal JSON string
```

Reply:
Added. They both fail now due to some logic added checking for leading whitespace.